src/celeritas/ext/detail/HitProcessor.cc

```diff
@@ -238,18 +274,24 @@ HitProcessor::~HitProcessor()
  */
 PrimaryId HitProcessor::register_primary(G4Track const& primary)
```

Comment:
This should also be mutable reference to indicate we're modifying the track by claiming ownership of its data.